### PR TITLE
offline mode for firewalled envs (part 2)

### PR DESCRIPTION
**Pull request by stas00**
In https://github.com/huggingface/transformers/pull/10407 I noticed I missed a few places where `local_files_only` should be overridden for the offline mode, so this PR completes the process.

Also rewrote the test to be more readable. Could test TF/Flax too but I don't have tiny models to run quick tests on.

@LysandreJik, @sgugger

### CODE REVIEW

**Comment by stas00** [comment]
Hmm, CI crashes once on 
```
try:
    nltk.data.find("tokenizers/punkt")
except LookupError:
```

https://app.circleci.com/pipelines/github/huggingface/transformers/20635/workflows/989fde0b-e543-4620-9d9a-f213ad53dd9b/jobs/176742
```
__________________ ERROR collecting examples/test_examples.py __________________
examples/test_examples.py:51: in <module>
    import run_seq2seq
examples/seq2seq/run_seq2seq.py:54: in <module>
    nltk.data.find("tokenizers/punkt")
../.local/lib/python3.6/site-packages/nltk/data.py:539: in find
    return FileSystemPathPointer(p)
../.local/lib/python3.6/site-packages/nltk/compat.py:41: in _decorator
    return init_func(*args, **kwargs)
../.local/lib/python3.6/site-packages/nltk/data.py:315: in __init__
    raise IOError("No such file or directory: %r" % _path)
E   OSError: No such file or directory: '/home/circleci/nltk_data/tokenizers/punkt/PY3'
```
which is odd, re-running the job fixed the problem.

But if it repeats we will simply change this to:

```
try:
    nltk.data.find("tokenizers/punkt")
except (LookupError, OSError):
```
not sure why the error was different here.

**Review by LysandreJik — APPROVED** [review]
Great, thanks for taking care of it.

File: examples/research_projects/bertabs/run_summarization.py [hunk 1]
```diff
@@ -9,8 +9,10 @@
 from torch.utils.data import DataLoader, SequentialSampler
 from tqdm import tqdm
 
+from filelock import FileLock
```
Comment (sgugger) [inline examples/research_projects/bertabs/run_summarization.py:12]: This file should not be touched. It is pinned to a previous version of Transformers in any case, so the offline mode won't work.

**Review by sgugger — APPROVED** [review]
Thanks for fixing those!

File: examples/research_projects/bertabs/run_summarization.py (hunk 1 above)
Comment (stas00) [reply examples/research_projects/bertabs/run_summarization.py]: ok, will restore it then. Thank you for the heads up, @sgugger

**Pull request merged**